app/controllers/api/callbacks/service_controller.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Api
+  module Callbacks
+    class ServiceController < ApplicationController
+      before_action :authorize
+      before_action :find_service
+
+      class Unauthorized < StandardError; end
+
+      rescue_from Unauthorized do |ex|
+        render json: { error: "Unauthorized Request: #{ex.message}" }, status: 401
+      end
+
+      def update
+        Rails.logger.info("Received DeployBot deploy callback. Updating #{params[:service_name]} with new ref #{params[:ref]}")
+        @service.update!(ref: params[:ref])
+        render json: { body: {} }, status: 200
+      end
+
+      private
+
+      def find_service
+        @service = Service.find_by!(name: params[:service_name])
+      rescue ActiveRecord::RecordNotFound
+        render json: { error: "Not Found" }, status: 404
+      end
+
+      def authorize
+        if (mac = request.authorization.presence)
+          digest = digest_request_parameters(InvocaSecrets['pre_deploy_checker', 'secret_key'])
```

Comment:
`config/dev_secrets.yml` needs to be updated to indicate this key is expected to exist. Also `secret_key` is very vague and doesn't give an idea of what it's needed for, `['pre_deploy_checker', 'api', 'auth_key']` would probably help indicate its purpose

Reply:
Also relative to the email about updating secrets, the key itself doesn't matter, you could use `openssl rand -base64 20` to come up with some random value and have SRE add that

Reply:
When we've got a value and the full key path name that we want, we can submit a secrets change request for production (since pre-deploy checker is in production) using these steps: https://ringrevenue.atlassian.net/wiki/spaces/DEV/pages/662406037/Updating+Using+Secrets#Updating&UsingSecrets-RequestingaChangefromtheSRETeam.

Reply:
thank you for all of those tips, followed all of that!